File: javascript/solution.test.js
const isAnagram = require('./solution');

describe('Anagram', () => {
  test('returns true for anagrams', () => {
    expect(isAnagram("listen", "silent")).toBe(true);
  });

  test('returns false for non-anagrams', () => {
    expect(isAnagram("hello", "world")).toBe(false);
    expect(isAnagram("abc", "def")).toBe(false);
  });

  test('handles case differences', () => {
    expect(isAnagram("Listen", "Silent")).toBe(true);
  });

  test('handles phrase anagrams', () => {
    expect(isAnagram("astronomer", "moon starer")).toBe(true);
  });
});

File: go/solution_test.go
package main

import "testing"

func TestIsAnagram(t *testing.T) {
	tests := []struct {
		s1       string
		s2       string
		expected bool
	}{
		{"listen", "silent", true},
		{"hello", "world", false},
		{"Listen", "Silent", true},
		{"astronomer", "moon starer", true},
		{"abc", "def", false},
	}

	for _, tt := range tests {
		result := IsAnagram(tt.s1, tt.s2)
		if result != tt.expected {
			t.Errorf("IsAnagram(%q, %q) = %v, want %v", tt.s1, tt.s2, result, tt.expected)
		}
	}
}
